### DISCUSSION TITLE
Using Keepassxc for Secret Service and Kwallet for other apps, is it possible?

### DISCUSSION THREAD

**Discussion by ioogithub** [keepassxreboot/keepassxc#8408] [Q&A]
Hi,

I have been successfully using keepassxc for secret service along side kwallet for 2 years . I used this guide: https://avaldes.co/2020/01/28/secret-service-keepassxc.html.

After a recent system update, this no longer works. Keepassxc now reports that another secret service is running (/usr/bin/kwalletd5).

If I kill kwalletd5 I lose the kwallet functionality which is essential as almost all KDE can only store passwords in kwallet but as long as kwallet is running I lost the Keepassxc secret service functionality.

Does anyone know how to get keepassxc secret service running along with kwallet? I believe the change must have come with added functionality of kwallet now acting as s secret service provider where is before I thought that KDE was still using some gnome apps for secret service ( gnome-keyring or pinentry-gnome3).

**Comment by ioogithub** [comment]
I found this bug report: https://www.mail-archive.com/kde-bugs-dist@kde.org/msg709199.html. I am not sure if it is the same issue I have or not, I want to link it here in case other users have the same problem. Perhaps someone with a better understanding of these systems could comment?

**Comment by droidmonkey** [comment]
Looks to be a bonefide bug in their software, the configuration item doesn't disable the service. A proposed patch fixes it: https://bugs.kde.org/show_bug.cgi?id=458069

**Reply by ioogithub** [reply]
This bug has been marked as revolved/fixed but only for archlinux. As I understand it, any distro. using KDE would be affected. This bug has wider ramifications and was marked a duplicate of this more active discussion: https://bugs.kde.org/show_bug.cgi?id=458085.

Unfortunately, it is quite a bit above my head however. Do you or other keepassxc developers who are familiar with how secret service works know if what they are recommending will allow keepassxc to be used with kwallet again?

The proposed solution seems to be to add "--no-allow-external-cache" to gpg-agent.conf. I tried this and I still can't get keepassxc working as a secret service agent again.

**Reply by droidmonkey** [reply]
There is nothing we can do to circumvent this bug without you uninstalling kwallet or side loading the patched version.

**Reply by michaelk83** [reply]
The fix for [Bug 458069](https://bugs.kde.org/show_bug.cgi?id=458069) is now [released in KDE Frameworks 5.98.0](https://kde.org/announcements/frameworks/5/5.98.0/#kwallet-framework).

The "--no-allow-external-cache" thing is a fix for a different (but related) problem, not what you need in your case. (Note that [Bug 458085](https://bugs.kde.org/show_bug.cgi?id=458085) turned out *not* to be a duplicate of 458069 after all. It's just that disabling the Secret Service integration in KWallet is one of the workarounds for 458085.)
